### ISSUE TITLE
Course order is creation order, not current order, in Purple Pen

### ISSUE THREAD

**Issue by tdobra** [tdobra/tctemplate#9]
The course order is recorded in the Purple Pen files using the order attribute, which is currently ignored. The courses are listed in order of creation in the PPen file, rather than their currently displayed order, which can lead to an unexpected ordering in TCTemplate.

**Comment by tdobra** [comment]
Fix deferred until after/as part of UI upgrade.

**Comment by mpickering** [comment]
Easy way to fix this is to sort `courseNodes` on the `order` attribute.

**Comment by mpickering** [comment]
```
diff --git a/WizardScript.js b/WizardScript.js
index 9529b67..8966a90 100644
--- a/WizardScript.js
+++ b/WizardScript.js
@@ -139,8 +139,10 @@ const tcTemplate = (() => {
             courseNodes[courseNodesId].parentNode.removeChild(courseNodes[courseNodesId]);
           }
         }
-        courseNodes = xmlobj.getElementsByTagName("course");
+        var courseNodesXML = xmlobj.getElementsByTagName("course");
         courseNodesNum = courseNodes.length;
+        courseNodes = Array.prototype.slice.call(courseNodesXML, 0);
+        courseNodes.sort(function (a, b) { return a.getAttribute("order") - b.getAttribute("order"); });
 
         //Make list of all course order attributes used - to determine print page, so must be completed before rest of file reading
         for (courseNodesId = 0; courseNodesId < courseNodesNum; courseNodesId++) {
```

**Comment by tdobra** [comment]
Thanks Matthew! I hadn't thought of your elegant way of fixing it. I've streamlined your approach some more before committing.
